lua/diffview/vcs/adapters/hg/init.lua

```diff
@@ -56,7 +56,7 @@ function HgAdapter.run_bootstrap()
   end
 
   local out = utils.job(vim.tbl_flatten({ hg_cmd, "version" }))
-  bs.version_string = out[1] and out[1]:match("Mercurial .*%(version (%S+)%)") or nil
+  bs.version_string = out[1] and out[1]:match("Mercurial .*%(version (%d%.%d%.%d).*%)") or nil
```

Comment:
This pattern won't match version numbers composed of more than a single digit.

```suggestion
  bs.version_string = out[1] and out[1]:match("Mercurial .*%(version (%d+%.%d+%.%d+)+%S*%)") or nil
```